test/algorithms/set_operations/intersection/alternative_side_strategy.hpp

```diff
@@ -0,0 +1,48 @@
+
+// Boost.Geometry (aka GGL, Generic Geometry Library)
+
+// Copyright (c) 2021 Barend Gehrels, Amsterdam, the Netherlands.
+
+// Licensed under the Boost Software License version 1.0.
+// http://www.boost.org/users/license.html
+
+
+#ifndef BOOST_GEOMETRY_TEST_INTERSECTION_ALTERNATIVE_SIDE_STRATEGY_HPP
+#define BOOST_GEOMETRY_TEST_INTERSECTION_ALTERNATIVE_SIDE_STRATEGY_HPP
+
+#include <boost/geometry/strategies/relate/cartesian.hpp>
+
+
+template <typename SideStrategy>
+class alternate_side_strategy
+    : public bg::strategies::envelope::cartesian<>
+{
+public:
+
+    using default_umbrella = bg::strategies::relate::cartesian<>;
+
+    template <typename Geometry1, typename Geometry2>
+    static auto disjoint(Geometry1 const& g1, Geometry2 const& g2)
+    {
+        return default_umbrella::disjoint(g1, g2);
+    }
+
+    template <typename Geometry1, typename Geometry2>
+    static auto relate(Geometry1 const& g1, Geometry2 const& g2)
+    {
+        return default_umbrella::relate(g1, g2);
+    }
+
+    static auto relate()
+    {
+        return default_umbrella::relate();
+    }
```

Comment:
This is in the tests so ok but there are some technicall issues I'd like to point out in case you wanted to build on top of it in the future.

1. If these functions are needed then this strategy should have been derived from `bg::strategies::relate::cartesian<>`. Then you could ommit these functions. Later if you needed to overload one of them you could bring the rest to the scope with e.g. `using bg::strategies::relate::cartesian<>::relate`.

2. `SideStrategy` may be incompatible with intersection and winding strategies. I'd implement base strategies for them taking `SideStrategy` as template argument and derive the public versions from these bases to hide this additional `SideStrategy` argument. Like it was done for spherical winding here:
https://github.com/boostorg/geometry/blob/develop/include/boost/geometry/strategies/spherical/point_in_poly_winding.hpp#L52